test/java/com/arpnetworking/metrics/portal/integration/repositories/JobExecutionRepositoryIT.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.integration.repositories;
+
+import com.arpnetworking.metrics.portal.TestBeanFactory;
+import com.arpnetworking.metrics.portal.scheduling.JobExecutionRepository;
+import com.google.common.base.Throwables;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import models.internal.Organization;
+import models.internal.scheduling.JobExecution;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Generic integration tests for implementations of {@link JobExecutionRepository}.
+ *
+ * @param <T> The job result type.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public abstract class JobExecutionRepositoryIT<T> {
```

Comment:
This is almost exactly what's in `ReportExecutionRepositoryIT`. You may just want to `git diff` against that.

I'm going to follow up this diff with a change to make that code use this instead, but the gist of it is `setUp` had to be fixed to be generic and the success tests needed a `newResult` callback since the result type became generic.